packages/context-menu/src/vaadin-context-menu-overlay.js

```diff
@@ -101,13 +102,32 @@ class ContextMenuOverlay extends OverlayElement {
     return {
       xMax: overlayRect.right - contentRect.width,
       xMin: overlayRect.left + contentRect.width,
-      yMax,
-      left: overlayRect.left,
-      right: overlayRect.right,
-      top: overlayRect.top,
-      width: contentRect.width
+      yMax
     };
   }
+
+  _updatePosition() {
+    this.style.transform = '';
+    super._updatePosition();
+
+    const isFullScreen = this.offsetWidth === window.innerWidth;
+    if (!isFullScreen && this.positionTarget && this.parentOverlay) {
+      // This overlay is positioned by a parent menu item,
+      // adjust the position by the overlay content paddings
+      const content = this.$.content;
+      const style = getComputedStyle(content);
+
+      const offsetY = this.hasAttribute('bottom-aligned')
```

Comment:
Would it make sense to make `super._updatePosition()` in `OverlayPositionMixin` return an object with changes it does, like `shouldAlignStartVertically` and `flexStart`? We could use them instead of checking for the `bottom-aligned`, `start-aligned` and `end-aligned` attributes to make the code cleaner.

Reply:
Changed the logic to not use the attributes. Would prefer not to change the position mixin in this PR if possible.